exonum/src/node/state.rs

```diff
@@ -754,6 +797,23 @@ impl State {
         full_proposes
     }
 
+    /// Checks whether some blocks are waiting for this transaction.
+    /// Returns a list of blocks that don't contain unknown transactions.
+    ///
+    /// Transaction is ignored if the following criteria are fulfilled:
+    ///
+    /// - transaction isn't contained in unknown transaction list of any blocks
+    /// - transaction isn't a part of block
+    pub fn remove_unknown_transaction(&mut self, tx_hash: Hash) -> Option<IncompleteBlock> {
+        if let Some(ref mut incomplete_block) = self.incomplete_block {
+            incomplete_block.unknown_txs.remove(&tx_hash);
+            if incomplete_block.unknown_txs.is_empty() {
+                return Some(incomplete_block.clone());
```

Comment:
Is cloning really required here? That is, can `self.incomplete_block` be used by other code after it is returned from this method?